File: backend/LogMin.Infrastructure/Persistence/Entities/Issue.cs
namespace LogMin.Infrastructure.Persistence.Entities;

public class Issue
{
    public Guid Id { get; set; }

    public string Pattern { get; set; } = default!;

    public string ServiceName { get; set; } = default!;

    public DateTime FirstSeen { get; set; }

    public DateTime LastSeen { get; set; }

    public int Count { get; set; }

    public double AvgScore { get; set; }

    public Guid RepresentativeLogId { get; set; }

    public bool IsAiProcessed { get; set; }

    public ICollection<Log> Logs { get; set; } = new List<Log>();

    public ICollection<IssueAnalysis> Analyses { get; set; } = new List<IssueAnalysis>();
}

File: frontend/logmind-dashboard/src/app/core/models/issue.ts
export interface IssueDto {
  id: string;
  pattern: string;
  serviceName: string;
  firstSeen: string;
  lastSeen: string;
  count: number;
  avgScore: number;
  isAiProcessed: boolean;
  representativeLogId: string;
}

export interface IssueLogReferenceDto {
  id: string;
  timestamp: string;
  message: string;
}

export interface IssueDetailsDto extends IssueDto {
  sampleLogs: IssueLogReferenceDto[];
  latestAnalysisId?: string | null;
}

export interface IssuesQuery {
  isAiProcessed?: boolean;
  serviceName?: string;
  pattern?: string;
  from?: string;
  to?: string;
  page?: number;
  pageSize?: number;
}
